### PR TITLE
Modernize codebase

### PR DESCRIPTION
**Pull request by djahandarie**
- Use ES modules
- Remove vendored libs and build them from npm using esbuild (except for handlebars-kbn, which is not available on npm and I have included in `dev/lib/handlebars`)
- Switch from JSZip to zip.js

### CODE REVIEW

**Comment by praschke** [comment]
`Logger` defined in `core.js` references `chrome`, which is a polyfill global that is set in `yomichan.js`. `yomichan.js` now includes from `core.js`. can globals be set in ES modules?

File: ext/js/app/content-script-wrapper.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+(async () => {
+    const src = chrome.runtime.getURL('js/app/content-script-main.js');
+    // eslint-disable-next-line no-unsanitized/method
+    const contentMain = await import(src);
+    contentMain.main();
```
Comment (praschke) [inline ext/js/app/content-script-wrapper.js:23]: no `main` function is defined in `content-script-main.js`

**Comment by praschke** [comment]
i submitted a PR against this branch addressing my review and Firefox issues. there are two more problems i found but i've run out of time for now:
- jszip was still used in tests, so those are currently broken
- eslint error: `setup()` is unused in `test/data/html/test-document2-script.js`

File: package.json [hunk 2]
```diff
@@ -36,19 +36,27 @@
...
         "eslint": "^8.52.0",
         "eslint-plugin-header": "^3.1.1",
         "eslint-plugin-jsdoc": "^46.8.2",
         "eslint-plugin-jsonc": "^2.10.0",
         "eslint-plugin-no-unsanitized": "^4.0.1",
         "fake-indexeddb": "^5.0.1",
+        "handlebars": "^4.7.8",
         "html-validate": "^8.7.0",
         "jsdom": "^22.1.0",
+        "jszip": "^3.10.1",
```
Comment (praschke) [inline package.json:56]: the tests haven't been moved from jszip to zip.js yet, but i think this dependency isn't currently being used

**Comment by praschke** [comment]
`legal.html` and the readme also don't reflect the change from jszip to @zip.js. additionally, as we're now building the extension dependencies from npm, referencing snapshots is no longer relevant.

**Comment by praschke** [comment]
LGTM. thank you for undertaking this slog.

File: package.json (hunk 2 above)
Comment (djahandarie) [reply package.json:57]: I think zip.js might be hard to use in node but I will take another shot in another PR at some point.

**Comment by Scrub1492** [comment]
Is there any reason for still using jszip as opposed to zip.js in tests?

**Comment by Scrub1492** [comment]
> Is there any reason for still using jszip as opposed to zip.js in tests?
@djahandarie

**Comment by djahandarie** [comment]
I tried to get zip.js to work but I couldn't get it working in Node for some reason. It certainly would be better to unify if possible.

**Comment by Scrub1492** [comment]
thanks for the confirmation

**Pull request merged**